sys/src/libacpi/harvey.c

```diff
@@ -436,22 +436,65 @@ acpihandler(void *_, void *arg)
 }
 #endif
 
+static struct handler ihandler;
+static Rendez irendez;
+static QLock irlock;
+static int intrpid;
+
+/*
+ * The acpica service routines can't run in the note handler
+ * because of a floating point exception. Fixable?
```

Comment:
I've cherry-picked these two commits and will be trying to get them ready in accordance with my review. This stuff is too good not to bring in.